src/middleware/passport-auth.ts

```diff
@@ -0,0 +1,168 @@
+    return createPrivateKey({ key: Buffer.from(privateKey, 'base64'), type: 'pkcs8', format: 'der' });
+};
+
+export const initPassport = async (userRepository: Repository<User>): Promise<void> => {
+    passport.use(
+        'jwt',
+        new JWTStrategy(
+            {
+                jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+                secretOrKey: process.env.JWT_SECRET!
+            },
+            async (jwtPayload, done) => {
+                logger.info('request authenticated with JWT');
+                const user = await userRepository.findOneBy({ id: jwtPayload?.user?.id });
+                return done(null, user);
+            }
+        )
+    );
+
+    if (process.env.AUTH_PROVIDERS?.includes('onelogin')) {
+        const oneLoginIssuer = await Issuer.discover(`${process.env.ONELOGIN_URL}/.well-known/openid-configuration`);
+        const privateKey = process.env.ONELOGIN_PRIVATE_KEY!.replace(/\\n/g, '\n');
+
+        passport.use(
+            'onelogin',
+            new OpenIdStrategy(
+                {
+                    client: new oneLoginIssuer.Client(
+                        {
+                            client_id: process.env.ONELOGIN_CLIENT_ID!,
+                            client_secret: process.env.ONELOGIN_CLIENT_SECRET!,
+                            redirect_uris: [`${process.env.BACKEND_URL}/auth/onelogin/callback`],
+                            token_endpoint_auth_method: 'private_key_jwt',
+                            token_endpoint_auth_signing_alg: 'PS256',
+                            id_token_signed_response_alg: 'ES256'
+                        },
+                        {
+                            keys: [readPrivateKey(privateKey).export({ format: 'jwk' })]
+                        }
+                    ),
+                    params: {
+                        response_type: 'code',
+                        ui_locales: 'en',
+                        scope: 'openid email'
+                        // TODO: we need to update our OneLogin config to be able to request the below claims
+                        // vtr: ['P2.Cl.Cm'],
+                        // claims: {
+                        //     userinfo: {
+                        //         'https://vocab.account.gov.uk/v1/coreIdentityJWT': null
+                        //     }
+                        // }
+                    }
+                },
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                async (tokenset: TokenSet, userInfo: UserinfoResponse, cb: any): Promise<void> => {
+                    logger.debug('auth callback from onelogin received');
+
+                    if (userInfo.email === undefined) {
+                        logger.error('onelogin auth failed: account has no email address');
+                        cb(null, undefined, {
+                            message: 'onelogin account does not have an email, use another provider'
```

Comment:
This shouldn't ever happen for OneLogin as far as I understand it, but might as well check.

Reply:
We need to... Maybe someones impersonating OneLogin maybe something has gone wrong.  We should always check fot the unlikely especially when we have no control over it.